src/compiler/transformers/esDecorators.ts

```diff
@@ -889,43 +889,63 @@ export function transformESDecorators(context: TransformationContext): (x: Sourc
             trailingBlockStatements = undefined;
         }
 
-        let newMembers: readonly ClassElement[] = members;
-
-        // insert a leading `static {}` block, if necessary
+        // prepare a leading `static {}` block, if necessary
+        let leadingStaticBlock: ClassStaticBlockDeclaration | undefined;
         if (leadingBlockStatements) {
+            // produces:
             //  class C {
             //      static { ... }
             //      ...
             //  }
             const leadingStaticBlockBody = factory.createBlock(leadingBlockStatements, /*multiLine*/ true);
-            const leadingStaticBlock = factory.createClassStaticBlockDeclaration(leadingStaticBlockBody);
+            leadingStaticBlock = factory.createClassStaticBlockDeclaration(leadingStaticBlockBody);
             if (shouldTransformPrivateStaticElementsInClass) {
                 // We use `InternalEmitFlags.TransformPrivateStaticElements` as a marker on a class static block
                 // to inform the classFields transform that it shouldn't rename `this` to `_classThis` in the
                 // transformed class static block.
                 setInternalEmitFlags(leadingStaticBlock, InternalEmitFlags.TransformPrivateStaticElements);
             }
-            newMembers = [leadingStaticBlock, ...newMembers];
-        }
-
-        // append the synthetic constructor, if necessary
-        if (syntheticConstructor) {
-            newMembers = [...newMembers, syntheticConstructor];
         }
 
-        // append a trailing `static {}` block, if necessary
+        // prepare a trailing `static {}` block, if necessary
+        let trailingStaticBlock: ClassStaticBlockDeclaration | undefined;
         if (trailingBlockStatements) {
+            // produces:
             //  class C {
             //      ...
             //      static { ... }
             //  }
             const trailingStaticBlockBody = factory.createBlock(trailingBlockStatements, /*multiLine*/ true);
-            const trailingStaticBlock = factory.createClassStaticBlockDeclaration(trailingStaticBlockBody);
-            newMembers = [...newMembers, trailingStaticBlock];
+            trailingStaticBlock = factory.createClassStaticBlockDeclaration(trailingStaticBlockBody);
         }

```

Comment:
```suggestion
        // produces:
        //  class C {
        //      ...
        //      static { ... }
        //  }
        const trailingStaticBlock = trailingBlockStatements
            && factory.createClassStaticBlockDeclaration(factory.createBlock(trailingBlockStatements, /*multiLine*/ true));
```

Reply:
I felt the `if` was easier to read, but I can make this change.

Reply:
I don't have strong feelings about it, I just thought the nested factory calls were easier to read and swept up the `if` in the change because that's how I would write it.